ee/codegen/src/generators/base-persisted-file.ts

```diff
@@ -17,6 +20,128 @@ export declare namespace BasePersistedFile {
   }
 }
 
+const VELLUM_MODULES = new Set(["vellum", "vellum_ee"]);
+
+class ImportSortedPythonFile extends PythonFile {
+  writeImports({
+    writer,
+    uniqueReferences,
+  }: {
+    writer: Writer;
+    uniqueReferences: Map<
+      string,
+      { modulePath: ModulePath; references: Reference[] }
+    >;
+  }) {
+    const sortedReferences = this.sortReferences(uniqueReferences);
+    let lastPriority: number | null = null;
+
+    for (const {
+      fullyQualifiedPath,
+      modulePath,
+      references,
+      priority,
+    } of sortedReferences) {
+      const refModulePath = modulePath;
+      // Check to see if the reference is defined in this same file and if so, skip its import
+      if (this.isDefinedInFile(refModulePath)) {
+        continue;
+      }
+
+      if (priority !== lastPriority) {
+        if (lastPriority !== null) {
+          writer.newLine();
+        }
+        lastPriority = priority;
+      }
+
+      if (refModulePath[0] === this.path[0]) {
+        // Relativize the import
+        // Calculate the common prefix length
+        let commonPrefixLength = 0;
+        while (
+          commonPrefixLength < this.path.length &&
+          commonPrefixLength < refModulePath.length &&
+          this.path[commonPrefixLength] === refModulePath[commonPrefixLength]
+        ) {
+          commonPrefixLength++;
+        }
+        // Calculate the number of levels to go up
+        let levelsUp = this.path.length - commonPrefixLength;
+        // If this is an __init__.py file, then we must go one more level up.
+        if (this.isInitFile) {
+          levelsUp++;
+        }
+        // Build the relative import path
+        let relativePath = levelsUp > 0 ? ".".repeat(levelsUp) : ".";
+        relativePath += refModulePath.slice(commonPrefixLength).join(".");
+        // Write the relative import statement
+        writer.write(
+          `from ${relativePath} import ${references
+            .map((reference) => this.getImportName({ writer, reference }))
+            .join(", ")}`
+        );
+      } else {
+        // Use fully qualified path
+        writer.write(
+          `from ${fullyQualifiedPath} import ${references
+            .map((reference) => this.getImportName({ writer, reference }))
+            .join(", ")}`
+        );
+      }
+      writer.newLine();
+    }
+    if (Object.keys(uniqueReferences).length > 0) {
+      writer.newLine();
+    }
+  }
+
+  // The above logic is copied and pasted from the Fern PythonFile AST Library
+  // The below is our only contribution to the class
```

Comment:
Once/If we fix Python File on the fern side, the sort logic we need is just 40 lines, not the entirety of isort. That's bc the vast majority of isort is:
- custom config - which we'll handle in userland not on our end
- ast building and writing - which codegen is already doing